PB-636: Responsive breakpoint switcher (#938)
magento/merchdocs: src/cms/page-builder-workspace.md

{% include page-builder-save-timeout.md %}

## Viewports

> A viewport is the visible area of a web page that a user sees. On the stage, when it is in full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.

Reviewer: ```suggestion
A _viewport_ is the visible area of a web page that a user sees. When it is in full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.
```

![Viewport buttons]({% link images/images-ee/page-builder-workspace-viewport-buttons.png %}){: .zoom}
_Viewport buttons_

> Page Builder also defines breakpoints to its viewports. Breakpoints define minimum and maximum widths within which certain styles are applied. Page Builder's two viewports define the following breakpoints for its content:

Reviewer: ```suggestion
Page Builder also defines breakpoints for viewports. Breakpoints define minimum and maximum widths within which certain styles are applied. The Page Builder viewports provide the following content breakpoints:
```

- **Desktop breakpoint**—`min-width: 1024px`. This breakpoint applies styles defined for viewport widths that measure 1024px and wider.
- **Mobile breakpoints**—`max-width: 768px, min-width: 640px`. These breakpoints apply styles defined for viewport widths between 768px and 640px.

Page Builder viewports provide two features: **content previews** and **breakpoint settings**.

> ### Content previews

Reviewer: Empty line after heading

```suggestion
### Content previews

```

By default, Page Builder provides two viewport previews:

> - **Desktop** — Previews your content without a predefined width. Desktop-defined styles (using breakpoint minimum width of 1024px) are still applied to the page. But the Desktop viewport width is defined by settings for container content types, like Rows. Selecting the Desktop viewport shows how your content will be styled on the storefront when the browser's page width is 1024px and wider.

Reviewer: ```suggestion
- **Desktop** — Displays the content preview without a predefined width. Desktop-defined styles (using breakpoint minimum width of 1024px) are still applied to the page. But the Desktop viewport width is defined by settings for container content types, like Rows. Selecting the Desktop viewport shows how your content will be styled on the storefront when the browser page width is 1024px and wider.
```

> - **Mobile** — Previews your content at a predefined width of 768px. Unlike the Desktop viewport, the Mobile viewport does show your page content at a width of 768px, along with the styles defined for the breakpoint widths of 768px (max) and 640px (min).

Reviewer: ```suggestion
- **Mobile** — Displays the content preview at a predefined width of 768px. Unlike the Desktop viewport, the Mobile viewport does show your page content at a width of 768px, along with the styles defined for the breakpoint widths of 768px (max) and 640px (min).
```

> The following screenshots show Page Builder's default responsive settings for product carousels displayed for these viewports:

Reviewer: ```suggestion
The following examples show the Page Builder default responsive settings for product carousels displayed for these viewports:
```

![Desktop viewport preview]({% link images/images-ee/page-builder-workspace-viewport-desktop.png %}){: .zoom}
_Desktop viewport preview, minimum width: 1024px_

![Mobile viewport preview]({% link images/images-ee/page-builder-workspace-viewport-mobile.png %}){: .zoom}
_Mobile viewport preview, width: 768px_

> ### Breakpoint settings

Reviewer: Empty line after heading

```suggestion
### Breakpoint settings

```

> The viewport buttons also provide the option to apply different breakpoint styles to content types based on the selected viewport. By default, Page Builder provides breakpoint settings for the Minimum Height fields of Rows, Columns, Tabs, Tab Items, Banners, Sliders, and Slides. When you select the Mobile viewport and then open the editor for one of those content types, you can enter field values specific to the Mobile viewport. Content type fields that allow specific breakpoint settings show an icon to the right of the field, as shown here for the Row:

Reviewer: ```suggestion
The viewport buttons also provide the option to apply different breakpoint styles to content types based on the selected viewport. By default, Page Builder provides breakpoint settings for the _Minimum Height_ settings of Rows, Columns, Tabs, Tab Items, Banners, Sliders, and Slides. When you select the Mobile viewport and then open the editor for one of those content types, you can enter values specific to the Mobile viewport. Content type fields that allow specific breakpoint settings display an icon to the right of the field, similar to the following example for a Row:
```

{% include page-builder-save-timeout.md %}

## Viewports

> A viewport is the visible area of a web page that a user sees. On Page Builder's stage, when it is in Page Builder's full-screen design mode, Page Builder displays viewport buttons above the stage to show you the content as the site user will see it on the storefront.

Author: Yes, viewport is a standard term.
